File: android/app/src/main/java/com/izzatismail/midtrim/domain/entity/ProjectInfo.kt
package com.izzatismail.midtrim.domain.entity

data class ProjectInfo(
    val id: String,
    val name: String,
    val trimDuration: Double,
    val wasCustomDuration: Boolean,
    val outputVideoUri: String,
    val thumbnailUri: String,
    val exportQualityTier: String,
    val mergedDuration: Double,
    val videoCount: Int,
    val createdAt: Long,
    val updatedAt: Long
)

data class SourceVideoInfo(
    val id: String,
    val projectId: String,
    val sourceVideoUri: String,
    val sourceVideoDuration: Double,
    val sourceFileSize: Long?,
    val orderIndex: Int,
    val trimStartTime: Double,
    val trimEndTime: Double
)

File: ios/MidTrim/MidTrim/Domain/Entities/ProjectInfo.swift
import Foundation

struct ProjectInfo: Equatable {
    let id: UUID
    let name: String
    let trimDuration: Double
    let wasCustomDuration: Bool
    let outputVideoURI: String
    let thumbnailURI: String
    let exportQualityTier: String
    let mergedDuration: Double
    let videoCount: Int
    let createdAt: Date
    let updatedAt: Date
}

struct SourceVideoInfo: Equatable {
    let id: UUID
    let projectId: UUID
    let sourceVideoURI: String
    let sourceVideoDuration: Double
    let sourceFileSize: Int64?
    let orderIndex: Int
    let trimStartTime: Double
    let trimEndTime: Double
}
